### ISSUE TITLE
detecting file type via extension is not a good idea

### ISSUE THREAD

**Issue by debugery** [Esenthel/EsenthelEngine#61]
you could use magic bytes, parse headers, etc

**Comment by GregSlazinski** [comment]
Where exactly is the problem and what issues are you experiencing?

**Comment by debugery** [comment]
For detecting media files as example for custom user specified playlists or traversing and detecting files in media directory or loading assets from inside of a directory i need to detect file format/codec/etc before loading them to Sound/Image/Video objects because supported codecs/types are very limited. Currently I'm using GetExt() or Flist-FileFind which are not enough.

**Comment by GregSlazinski** [comment]
There's: EXTENSION_TYPE ExtType(C Str &ext) to detect extension type.

If that's not enough then you should try:
-Image.Import
-SoundStream.create
-Video.create

They will already analyze file contents and try to open using all supported codecs.

To analyze file contents and tell what kind of file it is, without creating Image/SoundStream/Video object - that kind of function is not available in the engine.

**Comment by debugery** [comment]
Yes by GetExt() I mean ExtType() too. Thats the problem. I'm trying to traverse a lot files and loading them in those objects is not efficient. A simple static function in those structures name spaces to just detect is not hard to implement because the code already exist and runs with create/import. the difference is avoiding loading entire file and preparing entire structure for it.

**Comment by GregSlazinski** [comment]
I can reopen, but don't plan on doing that in the near future.

**Comment by debugery** [comment]
Thank you. You also could postpone full loading and do it on defined situations like playning the file or any other kind of access that require it, first step just store its name and other required data, full operation load and prepare it so i could scan extension, detect if supported and if fail fallback to full scan or just detect if supported without extension.
